modules/product/src/main/java/com/opengamma/strata/product/swap/InflationRateCalculation.java

```diff
@@ -176,8 +218,13 @@ public void collectIndices(ImmutableSet.Builder<Index> builder) {
   }
 
   // creates the rate observation
-  private RateObservation createRateObservation(SchedulePeriod period) {
+  private RateObservation createRateObservation(SchedulePeriod period, int scheduleIndex) {
+
+    // handle where index value at start date is known
     LocalDate endDate = period.getEndDate();
+    if (firstIndexValue != null && scheduleIndex == 0) {
```

Comment:
size > 1 case has to be excluded, e.g., by ArgChecker, rather than creating several accrual periods with only the fist period keeping the start index value?